picard/ui/statusindicator.py

```diff
@@ -17,37 +17,167 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
-from picard.const.sys import IS_WIN
+from picard.const.sys import (
+    IS_HAIKU,
+    IS_MACOS,
+    IS_WIN,
+)
 
 
 DesktopStatusIndicator = None
 
+
+class AbstractProgressStatusIndicator:
+    def __init__(self):
+        self._max_pending = 0
+        self._last_pending = 0
+
+    def update(self, files=0, albums=0, pending_files=0, pending_requests=0):
+        if not self.is_available:
+            return
+
+        total_pending = pending_files + pending_requests
+
+        if total_pending == self._last_pending:
+            return  # No changes, avoid update
+
+        previous_done = self._max_pending - self._last_pending
+        new_max = previous_done + total_pending
+        if new_max > self._max_pending:
+            self._max_pending = new_max
```

Comment:
`previous_done` and `new_max` intermediate variables are only used here, and while they make the code perhaps easier to understand, imho this code is simple enough to allow to replace those 4 lines by:

```python
self._max_pending = max(self._max_pending, self._max_pending - self._last_pending + total_pending)
```

Reply:
Good point. But getting rid of previous_done harms readability quite a bit, I changed it to:

```
previous_done = self._max_pending - self._last_pending
self._max_pending = max(self._max_pending, previous_done + total_pending)
```

I think that is quite readable.